src/Fixer/ClassNotation/ClassAttributesSeparationFixer.php

```diff
@@ -0,0 +1,395 @@
+class Sample
+{private $a; // a is awesome
+    /** second in a hour */
+    private $b;
+}
+                    ',
+                    ['elements' => ['property']]
+                ),
+                new CodeSample(
+                    '<?php
+class Sample
+{
+    const A = 1;
+    /** seconds in some hours */
+    const B = 3600;
+}
+                    ',
+                    ['elements' => ['const']]
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // Must run before BracesFixer and IndentationTypeFixer fixers because this fixer
+        // might add line breaks to the code without indenting.
+        return 55;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAnyTokenKindsFound(Token::getClassyTokenKinds());
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(SplFileInfo $file, Tokens $tokens)
+    {
+        $tokensAnalyzer = new TokensAnalyzer($tokens);
+        $class = $classStart = $classEnd = false;
+
+        foreach (array_reverse($tokensAnalyzer->getClassyElements(), true) as $index => $element) {
+            if (!isset($this->classElementTypes[$element['type']])) {
+                continue; // not configured to be fixed
+            }
+
+            if ($element['classIndex'] !== $class) {
+                $class = $element['classIndex'];
+                $classStart = $tokens->getNextTokenOfKind($class, ['{']);
+                $classEnd = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $classStart);
+            }
+
+            if ('method' === $element['type'] && !$tokens[$class]->isGivenKind(T_INTERFACE)) {
+                // method of class or trait
+                $attributes = $tokensAnalyzer->getMethodAttributes($index);
+
+                $methodEnd = true === $attributes['abstract']
+                    ? $tokens->getNextTokenOfKind($index, [';'])
+                    : $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $tokens->getNextTokenOfKind($index, ['{']))
+                ;
+
+                $this->fixSpaceBelowClassMethod($tokens, $classEnd, $methodEnd);
+                $this->fixSpaceAboveClassElement($tokens, $classStart, $index);
+
+                continue;
+            }
+
+            // `const`, `property` or `method` of an `interface`
+            $this->fixSpaceBelowClassElement($tokens, $classEnd, $tokens->getNextTokenOfKind($index, [';']));
+            $this->fixSpaceAboveClassElement($tokens, $classStart, $index);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition()
+    {
+        $types = ['const', 'method', 'property'];
+
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('elements', sprintf('List of classy elements; \'%s\'.', implode("', '", $types))))
```

Comment:
I'm not a big fan of this option, it's not flexible enough IMO. I would replace it with something like:
```php
[
    'const' => 'no_blank_line | single_blank_line | single_blank_line_if_phpdoc',
    'property' => 'no_blank_line | single_blank_line | single_blank_line_if_phpdoc',
    'method' => 'no_blank_line | single_blank_line | single_blank_line_if_phpdoc',
]
```
WDYT?

Reply:
I prefer to add one feature in one PR, here allow not _only_ methods but also other class elements to be separated by a single line. When this is done I'm happy to think about more options if there is a demand from a community for it ;)